lightning/src/ln/channel.rs

```diff
@@ -2471,36 +2514,32 @@ impl<Signer: Sign> Channel<Signer> {
 
-		let (num_htlcs, mut htlcs_cloned, commitment_tx, commitment_txid, feerate_per_kw) = {
-			let commitment_tx = self.build_commitment_transaction(self.cur_holder_commitment_transaction_number, &keys, true, false, logger);
-			let commitment_txid = {
-				let trusted_tx = commitment_tx.0.trust();
-				let bitcoin_tx = trusted_tx.built_transaction();
-				let sighash = bitcoin_tx.get_sighash_all(&funding_script, self.channel_value_satoshis);
-
-				log_trace!(logger, "Checking commitment tx signature {} by key {} against tx {} (sighash {}) with redeemscript {} in channel {}",
-					log_bytes!(msg.signature.serialize_compact()[..]),
-					log_bytes!(self.counterparty_funding_pubkey().serialize()), encode::serialize_hex(&bitcoin_tx.transaction),
-					log_bytes!(sighash[..]), encode::serialize_hex(&funding_script), log_bytes!(self.channel_id()));
-				if let Err(_) = self.secp_ctx.verify(&sighash, &msg.signature, &self.counterparty_funding_pubkey()) {
-					return Err((None, ChannelError::Close("Invalid commitment tx signature from peer".to_owned())));
-				}
-				bitcoin_tx.txid
-			};
-			let htlcs_cloned: Vec<_> = commitment_tx.3.iter().map(|htlc| (htlc.0.clone(), htlc.1.map(|h| h.clone()))).collect();
-			(commitment_tx.2, htlcs_cloned, commitment_tx.0, commitment_txid, commitment_tx.1)
+		let commitment_stats = self.build_commitment_transaction(self.cur_holder_commitment_transaction_number, &keys, true, false, logger);
+		let commitment_txid = {
+			let trusted_tx = commitment_stats.tx.trust();
+			let bitcoin_tx = trusted_tx.built_transaction();
+			let sighash = bitcoin_tx.get_sighash_all(&funding_script, self.channel_value_satoshis);
+
+			log_trace!(logger, "Checking commitment tx signature {} by key {} against tx {} (sighash {}) with redeemscript {} in channel {}",
+				log_bytes!(msg.signature.serialize_compact()[..]),
+				log_bytes!(self.counterparty_funding_pubkey().serialize()), encode::serialize_hex(&bitcoin_tx.transaction),
+				log_bytes!(sighash[..]), encode::serialize_hex(&funding_script), log_bytes!(self.channel_id()));
+			if let Err(_) = self.secp_ctx.verify(&sighash, &msg.signature, &self.counterparty_funding_pubkey()) {
+				return Err((None, ChannelError::Close("Invalid commitment tx signature from peer".to_owned())));
+			}
+			bitcoin_tx.txid
 		};
+		let mut htlcs_cloned: Vec<_> = commitment_stats.htlcs_included.iter().map(|htlc| (htlc.0.clone(), htlc.1.map(|h| h.clone()))).collect();
 
 		// If our counterparty updated the channel fee in this commitment transaction, check that
 		// they can actually afford the new fee now.
 		let update_fee = if let Some((_, update_state)) = self.pending_update_fee {
 			update_state == FeeUpdateState::RemoteAnnounced
 		} else { false };
 		if update_fee { debug_assert!(!self.is_outbound()); }
-		let total_fee = feerate_per_kw as u64 * (COMMITMENT_TX_BASE_WEIGHT + (num_htlcs as u64) * COMMITMENT_TX_WEIGHT_PER_HTLC) / 1000;
 		if update_fee {
```

Comment:
I think these two `if`s can b simplified to one

Reply:
Done.